<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>

</body>

</html>

<script>
    /*
        Throw, and Try...Catch...Finally
        The try statement defines a code block to run (to try). The catch statement defines a code block to handle any error. The finally statement defines a code block to run regardless of the result. The throw statement defines a custom error.
        Errors Will Happen!
        When executing JavaScript code, different errors can occur.
        Errors can be coding errors made by the programmer, errors due to wrong input, and other unforeseeable things.

        JavaScript try and catch
        The try statement allows you to define a block of code to be tested for errors while it is being executed.
        The catch statement allows you to define a block of code to be executed, if an error occurs in the try block.
        The JavaScript statements try and catch come in pairs:
    */

    document.write("<hr> Code Before Error");
    try {
        alerthdcjhjk("Viraj");
    } catch (err) {
        document.write("<hr>Error is " + err);
    }
    document.write("<hr> Code After Error");
</script>